src/utils/convert.js

```diff
@@ -338,13 +377,34 @@ const getStepMetaData = (definition, type, key) => {
   const hasPostScripting = hasScriptingMethod(js, type, key, 'post');
   const hasFullScripting = hasScriptingMethod(js, type, key, 'full');
 
-  const hasResourcePreScripting = js.indexOf(`${key}_pre_read_resource`) > 0;
-  const hasResourcePostScripting = js.indexOf(`${key}_post_read_resource`) > 0;
-  const hasResourceFullScripting = js.indexOf(`${key}_read_resource`) > 0;
+  const hasResourcePreScripting = hasScriptingMethod(js, type, key, 'pre', 'resource');
+  const hasResourcePostScripting = hasScriptingMethod(js, type, key, 'post', 'resource');
+  const hasResourceFullScripting = hasScriptingMethod(js, type, key, 'full', 'resource');
+
+  const hasInputFieldPreScripting = hasScriptingMethod(js, type, key, 'pre', 'input_fields');
+  const hasInputFieldPostScripting = hasScriptingMethod(js, type, key, 'post', 'input_fields');
+  const hasInputFieldFullScripting = hasScriptingMethod(js, type, key, 'full', 'input_fields');
+
+  const hasOutputFieldPreScripting = hasScriptingMethod(js, type, key, 'pre', 'output_fields');
+  const hasOutputFieldPostScripting = hasScriptingMethod(js, type, key, 'post', 'output_fields');
+  const hasOutputFieldFullScripting = hasScriptingMethod(js, type, key, 'full', 'output_fields');
```

Comment:
I've refactored here a bit so that we use `hasScriptingMethod()` to get all `hasXXXScripting` booleans. That means no direct calls to `js.indexOf(methodName)`.